### PR TITLE
Add configuration option for output format

### PR DESCRIPTION
**Pull request by jdkoren**
Adds a configuartion option to specify the output format for generated
docs, defaulting to 'html'. Hidden from help usage for the time being.

### CODE REVIEW

**Comment by devoncarew** [comment]
Can you elaborate on this PR a bit? Or, describe anticipated future output formats? I see you've wired up both `html` and `md`.

Adding @keertip for review and thoughts.

**Comment by devoncarew** [comment]
Ah, I see https://github.com/dart-lang/dartdoc/pull/1997 is a follow-up PR.

File: lib/src/dartdoc_options.dart [hunk 1]
```diff
@@ -1624,6 +1626,8 @@ Future<List<DartdocOption>> createDartdocOptions() async {
             'exist. Executables for different platforms are specified by '
             'giving the platform name as a key, and a list of strings as the '
             'command.'),
+    // TODO(jdkoren): unhide when this feature is in working order.
+    new DartdocOptionArgFile<String>('format', 'html', hide: true)
```
Comment (keertip) [inline lib/src/dartdoc_options.dart]: Does this have to be both command line and options file? Just have a command line flag only?

Comment (jdkoren) [reply lib/src/dartdoc_options.dart]: Done

**Pull request merged**